packages/node/src/common/AlternatingFileWriter.ts

```diff
@@ -0,0 +1,46 @@
+import fs from 'fs';
+
+export class AlternatingFileWriter {
+    private _fileStream?: fs.WriteStream;
+    private _count = 0;
+    private _disposed = false;
+
+    constructor(
+        private readonly _mainFile: string,
+        private readonly _fallbackFile: string,
+        private readonly _fileCapacity: number,
+    ) {
+        if (this._fileCapacity <= 0) {
+            throw new Error('File capacity may not be less or equal to 0.');
+        }
+    }
+
+    public async writeLine(value: string): Promise<this> {
+        if (this._disposed) {
+            throw new Error('This instance has been disposed.');
+        }
+
+        if (!this._fileStream) {
+            this._fileStream = fs.createWriteStream(this._mainFile, 'utf-8');
+        } else if (this._count >= this._fileCapacity) {
+            this._fileStream.close();
+            await fs.promises.rename(this._mainFile, this._fallbackFile);
+            this._count = 0;
+            this._fileStream = fs.createWriteStream(this._mainFile);
```

Comment:
I think it should have the same settings as a write stream from line 24

Reply:
Yeah, good catch. Changed